onnx/defs/parser.cc

```diff
@@ -54,8 +54,25 @@ Status ParserBase::Parse(Literal& result) {
     } else
       result.value = std::string(from + 1, next_ - from - 2); // skip enclosing quotes
   } else if ((isdigit(nextch) || (nextch == '-'))) {
+    bool could_be_inf = nextch == '-';
+
     ++next_;
 
+    // negative inf, has to be '-inf' now or nothing...
+    if (*next_ == 'i' && could_be_inf) {
+      ++next_;
+      if (*next_ == 'n' && next_ < end_) {
+        ++next_;
+        if (*next_ == 'f') {
+          ++next_;
+          result.value = std::string(from, next_ - from);
+          result.type = LiteralType::FLOAT_LITERAL;
+          return Status::OK();
+        }
+      }
+      return ParseError("'-inf' expected but not found.");
```

Comment:
What if a function is using a parameter whose name starts with `i` and different from ìnf`?

Reply:
Right, this is probably not the right way to go about fixing this. If this is indeed the way to go, as in allowing `{-}inf` and other non-numeric float values as valid literals.

Reply:
I think it would be helpful to have a more uniform approach, so that we can handle NaN also. Further, should this be case insensitive (just for compatibility with any standards out there)?

One possibility is to match an identifier, and then post-process the identifier with a map for special names like inf and NaN. (We'd have to do something similar for minus followed by an identifier.)

Reply:
Specifically, I think we could use something like:
```cpp
   while ((next_ < end_) && (isalpha(*next_))) {
      ++next_;
   }
   // Optionally check that that identifier is valid here, or better still, rely on
   // std::stof or std::stod to do the checking.
```

We should also change line 56 to allow `isalpha(nextch)`

Reply:
I've been trying to figure out an elegant way to achieve this, but it still escapes me. It seems like consuming isalpha characters like that runs into issues with the fact that you have to deal with the exponent notation when there are digits present. The parsing becomes very contextual.

Reply:
I am not sure I understand the concern with exponent notation ... that would be a problem if we wanted to support something like "e20" (an exponent without mantissa), but that's not allowed. The parsing is contextual, correct, for example, to distinguish plain identifiers from the use "inf" to mean a floating-point literal. We can support both (the use of "inf" as an identifier in an ONNX model, as well as its use as a literal) using context-based scanning and parsing. That is debatable, but orthogonal to the current issue.

Reply:
I would be happy to create a PR to update the parser to do this, if that would be helpful. This is a feature that would be nice and useful to have. We have a code-freeze coming up in another 3 weeks or so for the next release, so please let me know, thanks!

Reply:
I think this works now, thanks for your suggestions.